src/factories/flowRunArtifacts.ts

```diff
@@ -98,112 +101,28 @@ export async function flowRunArtifactsFactory() {
   }
 
   const checkLayout = throttle(async () => {
-    const visibleItems = [...artifacts.values()]
-    const availableClusterNodes = [...clusterNodes]
-    visibleItems.sort((itemA, itemB) => itemA.element.x - itemB.element.x)
+    availableClusterNodes = [...clusterNodes]
```

Comment:
Do we need to spread into a new array here? Looks like we're just looping over and calling render

Reply:
So `availableClusterNodes` is reset here so that as the collision check goes, it can grab a cluster or create a new one if there are none or they're all being used. When you zoom in and out, clusters can be created, then no longer needed. Rather than destroying them or creating another new one, they get reused.